tests/test_daily_temperature_extrema.py

```diff
@@ -0,0 +1,543 @@
+                        (DailyMaxTemperature, [3.0, 7.0]),
+                        (DailyMinTemperature, [1.0, 4.0])
+                    ])
+                    def it_returns_the_expected_extreme_values(query, DailyExtremeTemperature, statistics):
+                        results = query(DailyExtremeTemperature).order_by(DailyExtremeTemperature.obs_day)
+                        assert [r.statistic for r in results] == statistics
+
+                    @mark.parametrize('DailyExtremeTemperature', [DailyMaxTemperature, DailyMinTemperature])
+                    def it_returns_the_expected_data_coverages(query, DailyExtremeTemperature):
+                        results = query(DailyExtremeTemperature).order_by(DailyExtremeTemperature.obs_day)
+                        assert [r.data_coverage for r in results] == approx([3.0/24.0, 4.0/24.0])
+
+                def describe_with_many_observations_in_one_day_bis():
+                    '''Set up observations for native flag tests'''
+
+                    @fixture
+                    def obs_sesh(variable_sesh, var_temp_point, history_stn1_hourly):
+                        observations = [
+                            Obs(id=i, vars_id=var_temp_point.id, history_id=history_stn1_hourly.id,
+                                time=datetime.datetime(2000, 1, 1, i), datum=float(i))
+                            for i in range(0, 24)
+                            ]
+                        for sesh in generic_sesh(variable_sesh, observations):
+                            yield sesh
+
+                    # It would be better to DRY up describe_with_native_flags and describe_with_pcic_flags by
+                    # parametrizing, but pytest doesn't yet support parameterization over fixtures:
+                    # see https://github.com/pytest-dev/pytest/issues/349
+                    def describe_with_native_flags():
+                        '''2 native flags, 1 discard, 1 not discard'''
+
+                        @fixture
+                        def flag_sesh(obs_sesh, native_flag_discard, native_flag_non_discard):
+                            for sesh in generic_sesh(obs_sesh, [native_flag_discard, native_flag_non_discard]):
+                                yield sesh
+
+                        def describe_with_native_flag_associations():
+                            '''m < n associations of discard native flag to observations;
+                            k < n associations of not discard native flag to observations'''
+
+                            @fixture
+                            def flag_assoc_sesh(flag_sesh, native_flag_discard, native_flag_non_discard):
+                                sesh = flag_sesh
+                                obs = sesh.query(Obs)
+                                for id in range(0, 12):
+                                    obs.filter_by(id=id).first().native_flags.append(native_flag_discard)
```

Comment:
Ah, but you were talking about the (possible) retrieval of the objects from the database, yes? In the case that they were not already loaded, this would be better. But I think in this case, they are already in the session, and so retrieval is not an issue. Aren't these things fun to reason out.